Updates to node1

calvingsmith opened the pull request:
- remove google rhui client too
- installed some packages on node to support web console (these should be added to image)
- installed satellite host tools (don't add these to image)
- enabled remote execution 
- fixed satellite repo name
- re-enabled env flag to align with lifecycle environments

Also, request that the satellite image (don't know how to log into instance) get's the following run on it:

` satellite-installer --enable-foreman-plugin-remote-execution-cockpit`

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update RedHat8.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update setup.yml

calvingsmith committed:
Update RedHat8.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update server_register.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml:
```diff
@@ -41,6 +43,7 @@
   redhat_subscription:
     state: present
     activationkey: "{{ activation_key }}"
+    consumer_name: 
```
What does this do? Does it need a variable associated with it?

calvingsmith replied:
whoops! Yeah this argument allows you to change the name the system registers as.

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml:
```diff
@@ -51,3 +54,51 @@
     state: enabled
+
+#These should be included in the image
+- name: Install cockpit
```
Regarding cockpit, let's favor the [system role](https://console.redhat.com/ansible/automation-hub/repo/published/redhat/rhel_system_roles/content/role/cockpit) when available. Additionally, this should be its own playbook (we can use the existing system roles playbook) rather than an explicit part of the registration process.

calvingsmith replied:
OK have updated code to reflect that. Added the system role in the server_register.yml

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml:
```diff
@@ -51,3 +54,51 @@
+  when: ansible_distribution_major_version=="8"
+
+# Not these packages
+- name: install satellite client
```
I thought these were being deprecated.

calvingsmith replied:
Not these. The goferd and katello-agent are being deprecated. katello-host-tools has the yum plugins to update the satellite after package updates.

katello-host-tools-tracer enables a feature which tells the satellite after a package update if the system needs to be rebooted or a service restarted or a session reloaded.

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml:
```diff
@@ -51,3 +54,51 @@
+    
+- name: enable remote execution
+  authorized_key:
+    user: root
```
I don't think we should be encouraging root ssh as a best practice. could this key be added to "{{ ansible_user }}"?

calvingsmith replied:
I can make that change

willtome commented on satellite/server_register.yml:
```diff
@@ -2,7 +2,7 @@
   become: yes
   vars:
-    env: undef
+    # env: undef
```
Why comment this out?

calvingsmith replied:
This is in preparation for the 3 nodes when we get that arch  :-)

calvingsmith committed:
replaced root with {{ ansible_user }}

calvingsmith committed:
Update server_register.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Update main.yml

calvingsmith committed:
Set cockpit package load to minimal

calvingsmith committed:
Update RedHat8.yml

calvingsmith committed:
Set nodename in satellite UI

calvingsmith committed:
added rex_user variable

until I get around to re-configuring the satellite server to use a non-priviledged account

calvingsmith committed:
changed variable name

calvingsmith commented:
OK finished up the review comments. Switched to rex_user instead of ansible_user as need to configure satellite as well to get it to work right, will leave that as an issue if it's OK.

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/vars/RedHat8.yml:
```diff
@@ -3,3 +3,6 @@ rhsm_enabled_repos:
   - rhel-8-for-x86_64-baseos-rpms
   - rhel-8-for-x86_64-appstream-rpms
   - satellite-client-6-for-rhel-8-x86_64-rpms
+instance_name: "{{ inventory_hostname | regex_replace('_','-') }}"
```
These variables should go in defaults/main.yml so they will be defaults for RHEL 7 and 8.

I like this approach. Can you add activation_key with these as well and remove the set_fact for activation_key in the tasks?

calvingsmith replied:
OK, moved

willtome commented on collections/ansible_collections/demo/satellite/roles/register_host/vars/RedHat8.yml:
```diff
@@ -3,3 +3,6 @@ rhsm_enabled_repos:
   - satellite-client-6-for-rhel-8-x86_64-rpms
+instance_name: "{{ inventory_hostname | regex_replace('_','-') }}"
+rex_user: root
+# rex_user: "{{ ansible_user }}"
```
+1 for rex_user defaulting to ansible_user. That would be a service account in the environment that is already used for ssh management. Users could override with `root` if they wanted to take a less secure approach.

calvingsmith replied:
I agree. Will move to defaults/main.yml

willtome commented on satellite/server_register.yml:
```diff
@@ -4,5 +4,7 @@
+    cockpit_packages: minimal
   roles:
     - demo.satellite.register_host
+    - redhat.rhel_system_roles.cockpit
```
I'd like to keep this separate from the registration process. Could you add a job template to linux/setup.yml that uses the existing system_roles playbook and specifies the role + variables in extra_vars on the job template?

willtome reviewed (changes requested):
@calvingsmith these are some great improvements. Just requesting a few more changes then I'll be happy to run through testing. 

Do you think we need a README for the register_host role?

calvingsmith committed:
Added main.yml to centralize a few variables

calvingsmith committed:
- moved web console to own playbook
- moved variables to new defaults variables file

calvingsmith committed:
- moved web console to own playbook
- moved variables to new defaults variables file

calvingsmith committed:
- Fixed filename for webconsole script

calvingsmith committed:
- Fixed filename of webconsole script

calvingsmith committed:
- webconsole fixes

calvingsmith committed:
.

calvingsmith committed:
.

calvingsmith committed:
.

willtome commented on satellite/setup_webconsole.yml:
```diff
@@ -0,0 +1,8 @@
+  become: yes
+  vars:
+    # env: undef
+    satellite_url: "{{ lookup('ansible.builtin.env', 'SATELLITE_SERVER') }}"
```
is the satellite url required for installing cockpit?

willtome reviewed (changes requested):
@calvingsmith as discussed...
1) let's remove the web console playbook from this PR
2) the activation key task (line 38) in `collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml` as it is now in the variables file.
3) remove `cockpit_packages` line from satellite/server_register.yml

I think we'll be ready to merge after these changes

calvingsmith commented:
cool, working on the web console playbook in the linux section rn

On Thu, Nov 17, 2022 at 12:16 PM willtome ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> @calvingsmith <https://github.com/calvingsmith> as discussed...
>
>    1. let's remove the web console playbook from this PR
>    2. the activation key task (line 38) in
>    collections/ansible_collections/demo/satellite/roles/register_host/tasks/main.yml
>    as it is now in the variables file.
>    3. remove cockpit_packages line from satellite/server_register.yml
>
> I think we'll be ready to merge after these changes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/RedHatGov/product-demos/pull/42#pullrequestreview-1184730021>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAV4GWH6NRSM5HKLMRVBZDWIZR5LANCNFSM6AAAAAAR7DMQLI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Calvin Smith | Senior Solutions Architect
Public Sector Division | Red Hat
***@***.***

RED HAT | TRIED. TESTED. TRUSTED.
Every single executive department in the U.S. federal government runs Red
Hat in their datacenters.
Find out why at Trusted | Red Hat <http://www.redhat.com/en/about/trusted>

calvingsmith committed:
- Created web console job template in linux section

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
Updated setup.yml

calvingsmith committed:
- deleted setup_webconsole.yml
- removed activation key task as it's in variables file now
- removed cockpit_packages line

calvingsmith committed:
- Removed LINUX / setup web console from satellite setup (it gets pulled in from linux demos)

willtome commented:
@calvingsmith awesome PR! Thank you for the collaboration and for helping to improve this community.
